src/operator/tensor/diag_op-inl.h

```diff
@@ -69,7 +87,21 @@ inline TShape DiagShapeImpl(const TShape& ishape, const nnvm::dim_t k) {
     s = 0;
   }
 
-  return TShape({s});
+  if (x1 > x2) {
+    std::swap(x1, x2);
+  }
+
+  int n_dim = static_cast<int>(ishape.ndim()) - 1;
```

Comment:
Can n_dim be 0?

Reply:
No. What is a 0-dimensional array?

Reply:
What if ishape.ndim()==1

Reply:
Ok, I see. You've already checked that on line 66.